packages/client/src/services/LoadCanaryConfigService.ts

```diff
@@ -27,22 +27,50 @@ export default class LoadCanaryConfigService {
 
   async loadCanaryFromTemplate(): Promise<void> {
     const currentUrl = window.location.href;
-    const pattern = new RegExp(/.+template=([\w.-]+)(.+)?/);
-    let templateName = '';
 
-    if (currentUrl.match(pattern) !== null) {
-      const templateNameMatcher = pattern.exec(currentUrl);
-      templateName = templateNameMatcher ? templateNameMatcher[1] : '';
+    if (this.isBlankConfig(currentUrl)) {
+      return;
+    }
+    try {
+      const templateName = this.getTemplateName(currentUrl);
       const fileName = `${templateName}.json`;
 
-      axios.get(`${process.env.PUBLIC_URL}/templates/${fileName}`).then(response => {
-        configEditorStore.setCanaryConfigObject(response.data);
-        trackEvent(EVENT.LOAD_CONFIG, {
-          source: 'template',
-          template: templateName,
-          name: safeGet(() => response.data.name).orElse('UNKNOWN')
-        });
+      const templateCanaryConfig = await this.fetchTemplateContent(fileName);
+      configEditorStore.setCanaryConfigObject(templateCanaryConfig);
+      trackEvent(EVENT.LOAD_CONFIG, {
+        source: 'template',
+        template: templateName,
+        name: safeGet(() => templateCanaryConfig.name).orElse('UNKNOWN')
       });
+    } catch (e) {
+      log.error('Failed to fetch canary config template');
+      alert(`Failed to fetch canary config template:\n${e}`);
+    }
+  }
+
+  isBlankConfig(url: string): boolean {
+    const pattern = new RegExp(/.+config\/edit$/);
```

Comment:
Would these be more efficient if they were constants defined outside the methods?

Reply:
Good call out. I looked into JS regular expressions, and it looks like they are more efficient if I use the literal syntax instead of the constructor syntax for immutable regexs because the constructor syntax creates a new object every time.

As far as for increasing efficiency by defining them outside the methods, I am not sure.